modules/imap/functions.php

```diff
@@ -743,7 +743,8 @@ function merge_imap_search_results($ids, $search_type, $session, $hm_cache, $fol
         $imap = Hm_IMAP_List::connect($id, $cache);
         if (imap_authed($imap)) {
             $server_details = Hm_IMAP_List::dump($id);
-            $folder = $folders[$index];
+            $folder = $folders[$index] ?? $folders[0];
```

Comment:
We are no longer sending 2 request, i meant getting folder using loop $ids is not correct, the param should no longer be an array, any idea @kroky ?

Reply:
but I had an idea, if we send several folders at the same time especially folders it is already a table. like we want to display everything (send, received, etc...) which has as tag "Demo" for example